src/main/java/io/temporal/samples/hello/HelloPeriodic.java

```diff
@@ -88,14 +94,35 @@
   // Define the workflow implementation which implements the greetPeriodically workflow method.
   public static class GreetingWorkflowImpl implements GreetingWorkflow {
 
+    // In this example, we use an internal of 5 seconds with an intended variation of +/- 1 second
+    // between executions of some useful work. In real applications a higher value may be more
+    // appropriate, for example one that matches a business cycle of several hours or even days.
+    private static class ScheduleConfig {
+      static final int PeriodTargetSecs = 5;
+      static final int PeriodVariationSecs = 2;
+    }
+
+    // The max history length of a single Temporal workflow is 50,000 commands.
+    // Therefore, a workflow cannot we cannot run indefinitely. Instead, we use
+    // the ContinueAsNew feature to flow the logical execution thread to a new
+    // workflow run instance (same approach is used by Temporal's Cron-style
+    // scheduling system as well). In real life, the complexity of the workflow
+    // affects when we need to flow to a new run: I.e. if the workflow uses more
+    // commands per iteration, then we can perform fewer iterations before we must
+    // ContinueAsNew. For a simple workflow such as this, we could perform many
+    // thousands of iterations. However, for demonstration purposes we will flow
+    // to a new run more frequently.
+    // More details: https://docs.temporal.io/docs/java/workflows/#large-event-histories
+    private static final int SingleWorkflowRunIterations = 10;
+
     // Here we introduce a random delay between periodic executions
     private final Random random = Workflow.newRandom();
```

Comment:
Add a comment that Workflow.newRandom has to be used for determinism.

Reply:
✔️